app/services/notification_service.rb

```diff
@@ -42,4 +42,23 @@ def self.notify_start(event)
                                            30.minutes)
     end
   end
+
+  # Delete all notifications for a specific event
+  # This method will bypass callbacks to increase performance
+  def self.destroy_for(event)
+    notifications = Notification.where(notifyable: EventUser.where(event: event))
+    return if notifications.blank?
+
+    users = User.find(notifications.distinct.pluck(:user_id))
+
+    notifications.delete_all
+    update_counters(users)
+  end
+
+  # Updates the coutner caches
+  def self.update_counters(users)
+    users.each do |user|
```

Comment:
This could probably be done in a more efficient way:
```ruby
counts = Notification.join(:user).merge(users).group_by('users.id').select('users.id, count(*) as score')
counts.each do |c|
  User.update(c.id, notifications_count: c.score)
end
```

But perhaps then we will only get `N+1` instead of `2N + 1`.